dm/pkg/checker/utils.go

```diff
@@ -141,3 +147,38 @@ func isMySQLError(err error, code uint16) bool {
 	e, ok := err.(*mysql.MySQLError)
 	return ok && e.Number == code
 }
+
+// checkTables map schema => {table1, table2, ...}.
+// lackGrant map privilege => schema => table.
+func genExpectGrants(privileges map[pmysql.PrivilegeType]struct{}, checkTables map[string][]string) map[pmysql.PrivilegeType]map[string]map[string]struct{} {
+	lackGrants := make(map[pmysql.PrivilegeType]map[string]map[string]struct{}, len(privileges))
+	for p := range privileges {
+		lackGrants[p] = make(map[string]map[string]struct{}, len(checkTables))
+		for schema, tables := range checkTables {
+			if _, ok := lackGrants[p][schema]; !ok {
+				lackGrants[p][schema] = make(map[string]struct{}, len(tables))
+			}
+			for _, table := range tables {
+				lackGrants[p][schema][table] = struct{}{}
+			}
+		}
+		if p == pmysql.SelectPriv {
+			if _, ok := lackGrants[p]["INFORMATION_SCHEMA"]; !ok {
+				lackGrants[p]["INFORMATION_SCHEMA"] = make(map[string]struct{}, 1)
```

Comment:
```suggestion
				lackGrants[p]["INFORMATION_SCHEMA"] = make(map[string]struct{})
```

do we need to a `1` table to this map?